public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -431,3 +433,22 @@ export function getIntervals(range: TimeRange, lowLimit?: string, resolution?: n
 
   return rangeUtil.calculateInterval(range, resolution, lowLimit);
 }
+
+export function fixBothInstantAndRangeQuery(query: AlertQuery) {
+  const model = query.model;
+
+  if (!isPromQuery(model)) {
```

Comment:
I'm a bit concerned about distinguishing between data sources in the form. It might look like precedence to making data-source-specific transformations in the Alerting which I think we don't want to do?

Couldn't we move this transformation to the alert list panel `Create alert rule from this panel` button? The button has a link with the initial form values, maybe we could change the link instead of fixing it here?

Reply:
> Couldn't we move this transformation to the alert list panel Create alert rule from this panel button? The button has a link with the initial form values, maybe we could change the link instead of fixing it here?

Which of these two approaches results in simpler code? In either case we'll be special-casing Prometheus data sources for the purposes of alerting, or am I missing something?

Reply:
I share a similar concern, we can move it elsewhere but that would still make it choke on alert rules that somehow got in to the database with a `both` 🤔

Reply:
That's true. Let's keep it as it is :)